src/role_carry.js

```diff
@@ -399,8 +399,7 @@ roles.carry.action = function(creep) {
   // End of path, can't harvest, suicide (otherwise the sourcer gets stuck)
   if (!reverse && creep.body.filter((part) => part.type === WORK).length === 0) {
     // creep.log('Suiciding because end of path, no energy, do not want to get in the way of the sourcer (better recycle?)');
-    creep.memory.killed = true;
-    creep.suicide();
+    Creep.recycleCreep(creep);
```

Comment:
we should test this

Reply:
yes, this is what I worried about.
In worst cases might cause lots of carriers stuck together